src/uu/df/df.rs

```diff
@@ -0,0 +1,878 @@
+                Some(m)
+            }
+            _ => None,
+        }
+    }
+    #[cfg(windows)]
+    fn new(mut volume_name: String) -> Option<MountInfo> {
+        let mut dev_name_buf = [0u16; MAX_PATH];
+        volume_name.pop();
+        let dev_name_len = unsafe {
+            QueryDosDeviceW(
+                OsString::from(volume_name.clone())
+                    .as_os_str()
+                    .encode_wide()
+                    .chain(Some(0))
+                    .skip(4)
+                    .collect::<Vec<u16>>()
+                    .as_ptr(),
+                dev_name_buf.as_mut_ptr(),
+                dev_name_buf.len() as winapi::DWORD,
+            )
+        };
+        volume_name.push('\\');
+        let dev_name = LPWSTR2String(&dev_name_buf);
+
+        let mut mount_root_buf = [0u16; MAX_PATH];
+        let success = unsafe {
+            GetVolumePathNamesForVolumeNameW(
+                String2LPWSTR!(volume_name),
+                mount_root_buf.as_mut_ptr(),
+                mount_root_buf.len() as winapi::DWORD,
+                ptr::null_mut(),
+            )
+        };
+        if 0 == success {
+            // TODO: support the case when `GetLastError()` returns `ERROR_MORE_DATA`
+            return None;
+        }
+        let mount_root = LPWSTR2String(&mount_root_buf);
+
+        let mut fs_type_buf = [0u16; MAX_PATH];
+        let success = unsafe {
+            GetVolumeInformationW(
+                String2LPWSTR!(mount_root),
+                ptr::null_mut(),
+                0 as winapi::DWORD,
+                ptr::null_mut(),
+                ptr::null_mut(),
+                ptr::null_mut(),
+                fs_type_buf.as_mut_ptr(),
+                fs_type_buf.len() as winapi::DWORD,
+            )
+        };
+        let fs_type = if 0 != success {
+            Some(LPWSTR2String(&fs_type_buf))
+        } else {
+            None
+        };
+
+        let mut mn_info = MountInfo {
+            dev_id: volume_name,
+            dev_name,
+            fs_type: fs_type.unwrap_or("".to_string()),
+            mount_root,
+            mount_dir: "".to_string(),
+            mount_option: "".to_string(),
+            remote: false,
+            dummy: false,
+        };
+        mn_info.set_missing_fields();
+        Some(mn_info)
+    }
+}
+
+impl FsUsage {
+    #[cfg(unix)]
+    fn new(statvfs: libc::statvfs) -> FsUsage {
+        FsUsage {
+            blocksize: if statvfs.f_frsize != 0 {
+                statvfs.f_frsize as u64
+            } else {
+                statvfs.f_bsize as u64
+            },
+            blocks: statvfs.f_blocks as u64,
+            bfree: statvfs.f_bfree as u64,
+            bavail: statvfs.f_bavail as u64,
+            bavail_top_bit_set: ((statvfs.f_bavail as u64) & (1u64.rotate_right(1))) != 0,
+            files: statvfs.f_files as u64,
+            ffree: statvfs.f_ffree as u64,
+        }
+    }
+    #[cfg(not(unix))]
+    unimplemented!();
```

Comment:
```suggestion
    fn new(statvfs: libc::statvfs) -> FsUsage {
        unimplemented!();
    }
```
Bare `unimplemented()!` in an `impl` won't compile.